### PR TITLE
Remove the remaining Task Containers from SCDF

### PR DESCRIPTION
**Pull request by cppwfs**
This is the 3rd phase of removing the TaskContainers and allowing SCDF to use the Task Repositories And Task DAOs directly.   This PR also begins the migration of DAO's and Repositories from the AggregateTaskConfiguration to TaskConfiguration.   Eventually leading to the remove of AggregateTaskConfiguration.

This PR is spread across 5 commits based on a specific focus.

- 989036fca  Move TaskRepository and SCDFDao beans from AggregateTaskConfiguration to TaskConfiguration
- 2fbcb32b3 Replace TaskRepositoryContainer with TaskRepository
- 93e1b3c34 Replace DataflowTaskExecutionMetadataDaoContainer with DAO from dataflow
- d896f1856 Replace DataflowJobExecutionDaoContainer with dataflow's JobExecutionDao.
- f1aa305cd Replace DataflowTaskExecutionMetadataDaoContainer with DataflowTaskExecutionMetadataDao

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java [hunk 1]
```diff
@@ -211,6 +214,19 @@ public TaskAppDeploymentRequestCreator taskAppDeploymentRequestCreator(
 				metadataResolver, dataflowServerUri);
 	}
 
+	@Bean
+	public TaskRepository taskRepository(DataSource dataSource) {
+		MultiSchemaTaskExecutionDaoFactoryBean taskExecutionDaoFactoryBean = new MultiSchemaTaskExecutionDaoFactoryBean(dataSource, "TASK_");
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java:219]: So will the MultiSchema* components be removed when the aggregate is removed? Those were invented for the aggregate multi version support.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java [hunk 2]
```diff
@@ -169,7 +171,9 @@ private void createSampleJob(String jobName, int jobExecutionCount)
 		TaskExecution taskExecution = taskExecutionDao.createTaskExecution(jobName, LocalDateTime.now(), new ArrayList<String>(), null);
 
 		for (int i = 0; i < jobExecutionCount; i++) {
-			JobExecution jobExecution = jobRepository.createJobExecution(jobName, new JobParameters());
+			JobParameters jobParameters =
+				new JobParameters(Collections.singletonMap("parm", new JobParameter<>(i, Integer.class)));
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:175]: Is `parm` the correct key?

File: spring-cloud-dataflow-composed-task-runner/src/test/java/org/springframework/cloud/dataflow/composedtaskrunner/TaskLauncherTaskletTests.java [hunk 3]
```diff
@@ -524,7 +524,7 @@ TaskRepositoryInitializer taskRepositoryInitializer() {
 
 		@Bean
 		TaskExecutionDao taskExecutionDao(DataSource dataSource) {
-			return new JdbcTaskExecutionDao(dataSource);
+			return = new JdbcTaskExecutionDao(dataSource);
```
Comment (onobc) [inline spring-cloud-dataflow-composed-task-runner/src/test/java/org/springframework/cloud/dataflow/composedtaskrunner/TaskLauncherTaskletTests.java:527]: Is this a Java 17 feature? ;p

**Review by onobc — APPROVED** [review]
@cppwfs thanks,
A few minor comments to address but other than that LGTM

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java (hunk 1 above)
Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java:219]: Correct.   Working toward their retirement.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java (hunk 2 above)
Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:175]: Great question.    In this case we used to be able to create multiple job executions with a single instance for this command.   But that loophole was closed in Batch 5.  So now we need to make each JobExecution unique by either changing the name of the Job or adding a unique parameter value.   In our case we want the same job name, so we can add a unique job parameter.

Comment (onobc) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:175]: Oh ok, so this is in effect a "dummy param", correct? If so, maybe we should name the key something besides `parm` - maybe `unused` (or something like that). Wdyt?

Where does this show up as `parm`? In the UI?

Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:175]: Job Parameters can be viewed as a part of the Job Instance in the UI.   I can change it to test-parameter.

Comment (onobc) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/JobInstanceControllerTests.java:175]: Oh nevermind -  This is a test. Carry on.

**Comment by cppwfs** [comment]
Updated based on code review.  Squashed Merged.   Thanks @onobc !!!!

**Pull request merged**